pytensor/tensor/functional.py

```diff
@@ -0,0 +1,117 @@
+from typing import Callable, Optional
+
+from pytensor.graph import vectorize_graph
+from pytensor.tensor import TensorVariable
+from pytensor.tensor.utils import _parse_gufunc_signature
+
+
+def vectorize(func: Callable, signature: Optional[str] = None) -> Callable:
+    """Equivalent to numpy.vectorize.
+
+    If you have an existing graph you wish to vectorize, use `pytensor.graph.replace.vectorize_graph`
```

Comment:
Should this be a note further down below?

Reply:
Where would you put it?

Reply:
After Returns a new heading Notes

Reply:
I was going to comment on this too; it's just not so clear what it means (since a priori everything in pytensor is a graph).

Maybe:
```
Vectorize a python function with symbolic pytensor inputs and outputs according to a function signature, equivalent to numpy.vectorize.

Note that this function cannot operate on pytensor variables that represent computational graphs. To vectorize an existing pytensor graph, use `pytensor.graph.replace.vectorize`. 
```